### PR TITLE
2FA: Record activities for requested tokens

### PR DESCRIPTION
**Pull request by Betree**
Resolve https://github.com/opencollective/opencollective/issues/6536
Related: https://github.com/opencollective/opencollective/issues/6768

### CODE REVIEW

File: test/utils.js [hunk 1]
```diff
@@ -108,33 +106,25 @@ export const sleep = async (timeout = 200) =>
...
-      debugWaitForCondition(options.tag, `Has condition been met?`, hasConditionBeenMet);
-      if (hasConditionBeenMet) {
-        return setTimeout(resolve, options.delay || 0);
-      } else {
-        return setTimeout(isConditionMet, options.step || 100);
-      }
-    };
-    isConditionMet();
-  });
+export const waitForCondition = async (cond, options = { timeout: 10000, onFailure: null }) => {
```
Comment (Betree) [inline test/utils.js:113]: I needed a way to pass an async condition, took the opportunity to simplify the helper with async/await

File: server/constants/activities.ts [hunk 2]
```diff
@@ -103,6 +103,7 @@ enum ActivityTypes {
   OAUTH_APPLICATION_AUTHORIZED = 'oauth.application.authorized',
   TWO_FACTOR_CODE_ADDED = 'user.new.two.factor.code',
   TWO_FACTOR_CODE_DELETED = 'user.remove.two.factor.code',
```
[the remark this answers is not in the document]
Comment (znarf) [reply server/constants/activities.ts:105]: And add a comment sharing that we know the value of the constant is not ideal.

[the remark this answers is not in the document]
Comment (Betree) [reply server/constants/activities.ts:105]: I'll do it in a separate PR

File: test/utils.js (hunk 1 above)
Comment (znarf) [reply test/utils.js:114]: Nice one

**Review by znarf — APPROVED** [review]
Looks good.

File: server/constants/activities.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Betree) [reply server/constants/activities.ts:105]: to keep the diff clean

**Pull request merged**